### ISSUE TITLE
[BUG]: 

### ISSUE THREAD

**Issue by Tesla-Tank** [streamdeck-linux-gui/streamdeck-linux-gui#21]
### In raising this issue I confirm that

- [X] I have fully completed the issue template
- [X] I have searched open and closed issues for duplicates
- [X] I have read the [Contribution Guidelines](https://streamdeck-linux-gui.github.io/streamdeck-linux-gui/contributing/contributing-guide/)
- [X] I have read the [Code of Conduct](https://streamdeck-linux-gui.github.io/streamdeck-linux-gui/contributing/code-of-conduct/)
- [X] I have read the [Documentation](https://streamdeck-linux-gui.github.io/streamdeck-linux-gui/)

### Describe the bug

Using an edited gif as an icon can have a white screen appear when not in the original gif. I used gimp to change the frame speed, but somehow a white background appears in one of the frames. Both original and edited does not have a white background (Check screenshot for gifs).

### Steps to reproduce the behavior

1. Select an icon box 
2. Upload the edited image
3. See error

### Expected behavior

No white background frame appears.

### Screenshots

imgur links of the gifs used:
https://imgur.com/a/tUpmFAq

Video of example: (My apologies for the bad video quality)
https://github.com/timothycrosley/streamdeck-ui/assets/56700328/275a23f2-f169-4626-a6f2-43cb99b2c017
Edit: Left side is the edited gif and the right side is the original.

### System Information

 - OS: Nobara37 (I've also tried this with ubuntu mate (live usb) with the same bugged results.)
 - Desktop environment: KDE
 - Python version 3.11.3

### Stream Deck Version

Standard

**Comment by rkmax** [comment]
I had this problem when I started to use this app for the first time with gif customized by me, I still have no clear idea of the root cause of the issue since I need to investigate the logic behind the image filter but what I can tell you is avoid GIF with transparency since at the end the stream deck will not get advantage of that


my recommendation is to add some background color, choose wherever you want accordingly to your image/theme, here there's an example of a script that you may use, the dependencies are ffmpeg and convert (imagemagick)

```bash
#!/usr/bin/env bash

set -e

input=${1}


mkdir -p /tmp/frames

ffmpeg -i $input /tmp/frames/frame%04d.png

# replace background for the color that you want
for file in /tmp/frames//*.png; do
    convert "$file" -resize $size -background black -flatten "$file"
done

ffmpeg -i  .frames/frame%04d.png output.gif

```

**Comment by RemcoSchrijver** [comment]
I've seen similar behaviour in PNG Icons from the Material line where a white border was added to icons. Re-exporting it using GIMP made the issue go away. So this might be something in the image library that is used to create these images?

**Comment by rkmax** [comment]
> I've seen similar behaviour in PNG Icons from the Material line where a white border was added to icons. Re-exporting it using GIMP made the issue go away. So this might be something in the image library that is used to create these images?

@RemcoSchrijver could you please attach an image with the white border issue

**Comment by RemcoSchrijver** [comment]
@rkmax sorry for the delay but here you go, you can [download this icon from material icons](https://fonts.google.com/icons?selected=Material+Icons+Outlined:home:&icon.platform=web&icon.set=Material+Icons), and will see whjite ghosting around the image. Which is fixed when I let it be re-exported by GIMP. 

_The image causing the problems:_
![outline_home_white_48dp](https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/assets/45097990/8d497588-94c8-463c-97c3-9dfc92a3007f)

_How it actually looks like on the streamdeck:_
![8e3c8149-55b3-4cd5-a96e-28e6343f3322](https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/assets/45097990/47387942-09be-467c-a6a2-de366c77fbdb)

As you can see it looks a bit wonky, compared to the other icons next to it.

**Comment by rkmax** [comment]
@RemcoSchrijver thank you for your information.

I'll push another fix soon improving how the images are being handle that luckily will take care of this issue and others


Here's how your image looks after the fix

![WhatsApp Image 2023-10-24 at 2 34 29 PM](https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/assets/395707/9883dfb7-ea6e-449d-871d-feeaec4088e8)

**Comment by RemcoSchrijver** [comment]
> @RemcoSchrijver thank you for your information.
> 
> I'll push another fix soon improving how the images are being handle that luckily will take care of this issue and others
> 
> 
> Here's how your image looks after the fix
> 
> ![WhatsApp Image 2023-10-24 at 2 34 29 PM](https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/assets/395707/9883dfb7-ea6e-449d-871d-feeaec4088e8)
> 

Yeah that's also how it shows up when I re-export it with GIMP, so glad to hear this works!

**Comment by rkmax** [comment]
@RemcoSchrijver https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/pull/109 this PR coverts this Gifs with encoded palettes (the original fix) and other modes like grayscale with transparency (the image that you pass) among others modes that contains transparency
